### PR TITLE
fix(websockets): remove client when connection is closed

### PR DESCRIPTION
**Pull request by petrjasek**
in case connection was closed by client (and not lost) it is using different callback which was not handled before. plus heartbeat is now part of server factory

### CODE REVIEW

File: superdesk/ws.py [hunk 1]
```diff
@@ -54,10 +52,15 @@ def onMessage(self, payload, isBinary):
...
     def connectionLost(self, reason):
-        """Unregister on connection drop"""
+        """Unregister client on lost connection."""
         WebSocketServerProtocol.connectionLost(self, reason)
         self.factory.unregister(self)
 
+    def onClose(self, wasClean, code, reason):
+        """Unregister client on closed connection."""
+        WebSocketServerProtocol.onClose(self, wasClean, code, reason)
+        self.factory.unregister(self)
```
Comment (amagdas) [inline superdesk/ws.py]: Should we unregister before calling `onClose` and `connectionLost`? In case of an error being thrown it could remain registered.

**Comment by amagdas** [comment]
Looks good :+1: 
Not sure if my comment is a valid concern.

File: superdesk/ws.py (hunk 1 above)
Comment (petrjasek) [inline superdesk/ws.py]: i think those are only logging and if those would throw error the whole thing would stop anyhow

Comment (amagdas) [inline superdesk/ws.py]: OK

**Comment by ioanpocol** [comment]
:+1:

**Pull request merged**